staging/src/k8s.io/cli-runtime/pkg/resource/selector.go

```diff
@@ -53,6 +57,11 @@ func (r *Selector) Visit(fn VisitorFunc) error {
 		FieldSelector: r.FieldSelector,
 		Limit:         r.LimitChunks,
 	}
+
+	if os.Getenv("ENABLE_KUBECTL_WATCH_LIST_ALPHA") == "true" {
```

Comment:
We have a dedicated mechanism for these operations, see https://github.com/kubernetes/kubernetes/blob/1904a56e3071035de19016cef621866a37140d5f/staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go#L425-L448

Reply:
thanks, it's have been addressed.

Reply:
There is a cycle dependeny here if we import `util` pkg into `resource`, because pkg `util` dependent on `resource` currently. So I move `FeatureGate` from `util` into `util/featuregate` to avoid it.